components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2645,111 +2638,63 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
+                if (error.code === ErrorCodes.PERMISSION_DENIED) {
+                    return false;
+                } else {
+                    throw error;
+                }
             }
-
-            return result;
         };
-
+        if (fineGrainedOp === "not_implemented") {
+            // we ignore fine grained permissions if not implemented
+            return checkAgainstDB();
+        }
+        // run check against old DB logic and new permission system in parallel
         const [fromDB, fromCentralizedPerms] = await Promise.allSettled([
-            // Permission checks against the DB will throw, if the user is not permitted to perform the action, or if iteraction with
-            // dependencies (DB) fail.
             checkAgainstDB(),
-
-            // Centralized perms checks only throw, when an interaction error occurs - connection not available or similar.
-            // When the user is not permitted to perform the action, the call will resolve, encoding the result in the response.
-            checkWithCentralizedPerms(),
+            this.authPermissions.hasPermissionOnOrg(user.id, fineGrainedOp, org.id),
         ]);
 
-        // check against DB resolved, which means the user is permitted to perform the action
-        if (fromDB.status === "fulfilled") {
-            if (fromCentralizedPerms.status === "fulfilled") {
-                // we got a result from centralized perms, but we still need to check if the outcome was such that the user is permitted
-                reportCentralizedPermsValidation(fineGrainedOp, fromCentralizedPerms.value.permitted === true);
-            } else {
-                // centralized perms promise rejected, we do not have an agreement
-                reportCentralizedPermsValidation(fineGrainedOp, false);
-            }
-
-            // Always return the result from the DB check
-            return fromDB.value;
-        } else {
-            // The check agains the DB failed. This means the user does not have access.
-
-            if (fromCentralizedPerms.status === "fulfilled") {
-                // we got a result from centralized perms, but we still need to check if the outcome was such that the user is NOT permitted
-                reportCentralizedPermsValidation(fineGrainedOp, fromCentralizedPerms.value.permitted === false);
-            } else {
-                // centralized perms promise rejected, we do not have an agreement
-                reportCentralizedPermsValidation(fineGrainedOp, false);
-            }
-
-            // We re-throw the error from the DB permission check, to propagate it upstream.
+        // report what we got from centralized perms
+        if (fromCentralizedPerms.status === "fulfilled") {
+            reportCentralizedPermsValidation(fineGrainedOp, !!fromCentralizedPerms.value);
+        }
+        // if fromDB errored we throw here
+        if (fromDB.status === "rejected") {
             throw fromDB.reason;
```

Comment:
This removes the logic which reports whether the permissions matched the value from DB. How are we going to monitor if the checks agree?

Reply:
this is only the exceptional error case. See checkAgainstDB returns false if the permission doesn't exist.